lib/engines/content_object_store/app/services/content_object_store/create_edition_service.rb

```diff
@@ -8,7 +10,9 @@ def initialize(schema, edition_params)
     def call
       ActiveRecord::Base.transaction do
         content_block_edition = create_whitehall_edition
-        create_publishing_api_edition(content_id: content_block_edition.document.content_id)
+        content_id = content_block_edition.document.content_id
+        create_publishing_api_edition(content_id:)
+        publish_publishing_api_edition(content_id:)
```

Comment:
optional: The `publish_publishing_api_edition` method now also takes care of discarding drafts which isn't clear from this top level composer. It might be clearer to the reader to handle that here instead:

```ruby
def call
  ActiveRecord::Base.transaction do
    content_block_edition = create_whitehall_edition
    content_id = content_block_edition.document.content_id
    
    create_publishing_api_draft_edition(content_id: content_block_edition.document.content_id)
    publish_publishing_api_edition(content_id:)
  rescue => e
    discard_publishing_api_draft_edition(content_id:)
    raise e
  end
end
```

Reply:
Yeah, I thought about that, but we probably don't want to discard the draft if the draft creation has failed. I do agree it's clearer though

Reply:
Ah, but if I rescue from the `PublishingFailureError` method, that'll only recover from that error, so that should work